pkg/controller/nodeipam/ipam/controller.go

```
@@ -92,6 +92,9 @@ func NewController(
 		return nil, err
 	}
 
+	if _, err := c.set.AllocateNext(); err != nil {
```

don't you need to dealloc

---

Also, there should be log with error and I would return a new error type.

---

Done. There is no error log in the NewController function. Can I return a new error type without printing log?